crates/hir_ty/src/lib.rs

```diff
@@ -843,6 +848,33 @@ impl Ty {
 
     pub fn impl_trait_bounds(&self, db: &dyn HirDatabase) -> Option<Vec<GenericPredicate>> {
         match self {
+            Ty::Apply(ApplicationTy { ctor: TypeCtor::OpaqueType(opaque_ty_id), parameters }) => {
```

Comment:
why is this necessary?

Reply:
This method is used in type walking, which is used for "goto type" hover.
Seems that I did it in a wrong way before.
But now it's fixed and a test is added. It's required to provide trait "Future" in "goto type" hover.